code/modules/holodeck/basketball.dm

```diff
@@ -0,0 +1,103 @@
+//Basketball items and structures
+
+/obj/item/toy/basketball
+	name = "basketball"
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "basketball"
+	inhand_icon_state = "basketball"
+	desc = "Here's your chance, do your dance at the Space Jam."
+	w_class = WEIGHT_CLASS_BULKY //Stops people from hiding it in their bags/pockets
+
+
+/obj/structure/hoop
+	name = "basketball hoop"
+	desc = "Boom, shakalaka!"
+	icon = 'icons/obj/basketball.dmi'
+	icon_state = "hoop"
+	anchored = TRUE
+	density = TRUE
+
+/**
+* this proc is for dunking an object
+*
+* ball_exp is how much exp you get for a dunk, ball_skill is balling skill level modifier
+* dunking has a 60% chance of failure which decreases based on balling skill
+*			unsuccessful: 5 exp; 50 stamina damage; ball goes in a random direction one step away
+*			sucessful: 20 exp; 40-10 stamina damage (based on ball_skill); ball goes in net
+*/
+/obj/structure/hoop/attackby(obj/item/dunk_object, mob/user, params)
+	if(get_dist(src,user)>=2) //can only dunk when close
+		return
+
+	var/ball_skill = user.mind?.get_skill_modifier(/datum/skill/balling, SKILL_PROBS_MODIFIER)
+	var/ball_exp = 0
+
+	if(prob(60 - ball_skill)) //60% base chance to fail a dunk
+		if(isliving(user))
+			var/mob/living/living_baller = user
+			living_baller.apply_damage(50, STAMINA) //failing a dunk is universally taxing
+		ball_exp = 5
+		user?.mind.adjust_experience(/datum/skill/balling, ball_exp)
+		visible_message(span_danger("[user] fumbles the dunk!"))
+		user.dropItemToGround(dunk_object)
+		step(dunk_object, pick(NORTH,SOUTH,EAST,WEST))
+		return
+
+	if(user.transferItemToLoc(dunk_object, drop_location()))
+		var/stam_cost = 40 - ball_skill / 2
+		if(isliving(user))
+			var/mob/living/living_baller = user
+			living_baller.apply_damage(stam_cost, STAMINA)
+		ball_exp = 20
+		user?.mind.adjust_experience(/datum/skill/balling, ball_exp)
+		visible_message(span_warning("[user] dunks [dunk_object] into \the [src]!"))
+
+/obj/structure/hoop/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+	if(user.pulling && isliving(user.pulling))
+		var/mob/living/victim = user.pulling
+		if(user.grab_state < GRAB_AGGRESSIVE)
+			to_chat(user, span_warning("You need a better grip to do that!"))
+			return
+		victim.forceMove(loc)
+		victim.Paralyze(100)
+		visible_message(span_danger("[user] dunks [victim] into \the [src]!"))
+		user.stop_pulling()
+	else
+		..()
```

Comment:
This is going to call parent for a second time. Is this intended?

Reply:
I did not code that but I don't believe so, so I removed it. I tested and couldn't see any difference in behavior after.